neurodsp/sim.py

```diff
@@ -120,78 +99,89 @@ def sim_oscillator(n_samples_cycle, n_cycles, rdsym=.5):
     pha_one_cycle = np.hstack([np.linspace(
         0, np.pi, decay_samples + 1), np.linspace(-np.pi, 0, rise_samples + 1)[1:-1]])
     phase_t = np.tile(pha_one_cycle, n_cycles)
+    phase_t = phase_t[:n_samples]
 
     # Transform phase into an oscillator
-    oscillator = np.cos(phase_t)
+    osc = np.cos(phase_t)
 
-    return oscillator
+    return osc
 
 
-def sim_noisy_oscillator(freq, n_seconds, fs, rdsym=.5, f_hipass_brown=2, ratio_osc_power=1):
+def sim_noisy_oscillator(n_seconds, fs, freq, exponent=2, rdsym=.5, ratio_osc_power=1,
+                         f_range_filter=(2, None), filter_order=None):
     """Simulate an oscillation embedded in background 1/f.
 
     Parameters
     ----------
-    freq : float
-        Oscillator frequency
     n_seconds : float
         Signal duration, in seconds
     fs : float
         Signal sampling rate, in Hz
-    f_hipass_brown : float
-        Frequency (Hz) at which to high-pass-filter brown noise
+    freq : float
+        Oscillator frequency
+    exponent : float
+        Desired power-law exponent - beta in P(f)=f^beta
+    rdsym : float
+        Rise-decay symmetry of the oscillator, as fraction of the period in the rise time;
+            =0.5 - symmetric (sine wave)
+            <0.5 - shorter rise, longer decay
+            >0.5 - longer rise, shorter decay
     ratio_osc_power : float
-        Ratio of oscillator power to brown noise power
+        Ratio of oscillator power to noise power
             >1 - oscillator is stronger
             <1 - noise is stronger
+    f_range_filter : 2-element array (lo,hi) or None
+        Frequency range of simulated noise
+            if None: do not filter
+    filter_order : int
+        Order of filter for noise process
 
     Returns
     -------
-    1d array
-        Oscillator with brown noise
+    osc: 1d array
+        Oscillator with noise
     """
 
-    # Determine order of highpass filter (3 cycles of f_hipass_brown)
-    filter_order = int(3 * fs / f_hipass_brown)
-    if filter_order % 2 == 0:
-        filter_order += 1
+    # Determine order of highpass filter (3 cycles of f_hipass)
+    if f_range_filter is not None:
+        if filter_order is None:
+            filter_order = int(3 * fs / f_range_filter[0])
+        if filter_order % 2 == 0:
+            filter_order += 1
 
     # Determine length of signal in samples
     n_samples = int(n_seconds * fs)
 
-    # Generate filtered brown noise
-    brown = sim_filtered_brown_noise(n_seconds, fs, (f_hipass_brown, None), filter_order)
+    # Generate filtered noise
+    noise = sim_filtered_noise(n_seconds, fs, f_range_filter, filter_order, exponent=exponent)
```

Comment:
do we want to pass in the different noise sim functions, instead of constraining to filtered noise? Because as of now this cannot produce the plateau/knee in time domain, as FOOOF is able to do in freq-domain.